cogs/highlights.py

```diff
@@ -53,11 +69,13 @@ async def _send_highlights(
         self, con: QuoteBotDatabaseConnection, msg: discord.Message, highlights: Iterable[sqlite3.Row]
     ) -> None:
         seen_user_ids = set()
-        for user_id, query in highlights:
+        for user_id, query, guild_id in highlights:
             if not self.bot.get_user(user_id):
                 await con.clear_user_highlights(user_id)
             elif not (member := msg.guild.get_member(user_id)):
                 continue
+            elif guild_id and guild_id != msg.guild.id:
```

Comment:
The first condition (before the `and`) should be made more explicit (`guild_id > 0` or `guild_id is not None`)

Reply:
`guild_id` can only be an `int`, for which I would usually write `guild_id` instead of `guild_id != 0`. But ok, using `> 0`.